test/functional/page_objects/settings_page.ts

```diff
@@ -357,7 +361,12 @@ export class SettingsPageObject extends FtrService {
       }
 
       await this.header.waitUntilLoadingHasFinished();
-      await this.clickAddNewIndexPatternButton();
+      const flyOut = await this.testSubjects.exists('createAnyway');
```

Comment:
I haven't confirmed, but I think there's actually 3 possible cases when you go to index patterns;
1. You have existing index patterns and you go to the index pattern listing page with the "Create index pattern " button
2. You have data (that you have read access to) but no index patterns, you get one flyout
3. You don't have any non-system indices that you have access to, and you get a different flyout with the link like "go to index patterns anyway" (the `createAnyway` data-test-subj)

I think the code here handles two of those cases (1 and 3).  Ideally I think most of our tests should be starting from a clean state, load data, and go to index patterns.  Which I think is case 2 above.  But I know this test passes locally when run by itself with a clean starting point.  And I know it passes in CI.  So I think this PR is probably OK for now.

Reply:
maybe someone from @elastic/kibana-app-services could help explain the flow here?